### PR TITLE
test-all-scream: Move baseline build dir

### PR DESCRIPTION
**Pull request by jgfouca**
Doing it in the baseline dir seems like a bad idea. One, potential races with other TAS runs. Two, on some machines like pmgpu, the baseline filesystem is MUCH slower than the scratch area.

### CODE REVIEW

File: components/eamxx/scripts/test_all_scream.py
Comment (bartgol) [inline components/eamxx/scripts/test_all_scream.py:892]: If I understand correctly, the location of SCREAM_DATA_DIR when generating baselines is in the baseline dir, while the actual code is built and run in the usual ctest-build subdir, right?

If that's the case, I would _not_ clean up the folder at the end. We already nuke them at the beginning of the test phase anyways. The advantage of not nuking them is that, if something fails, we can cd in the folder and see what was going on (e.g., inspect the cache, check flags, etc.).

**Review by bartgol — COMMENTED** [review]
Assuming I understood what this PR does, I like it. I have a suggestion though, so I'm holding off approval to avoid a possibly premature automerge. I'll approve after we settle that.

File: components/eamxx/scripts/test_all_scream.py
Comment (jgfouca) [reply components/eamxx/scripts/test_all_scream.py:892]: @bartgol , I had to move the "make test dirs and nuke them" call earlier in TAS in order to ensure this directory existed for generate_baseline. I don't think they get nuked again, so leaving the files around could case problems with the test build. That said, I don't think it would be hard to put the baseline build in a subdirectory that would be harmless to the main build. Something like `ctest-build/debug/baseline_build`.

File: components/eamxx/scripts/test_all_scream.py
Comment (bartgol) [inline components/eamxx/scripts/test_all_scream.py:896]: We may need to add one of these also _before_ building baselines. I'm thinking of this scenario:

1. run tas: baseline build fails, so we don't nuke the dir
2. change something, run tas again: baseline dir is not clean, so we may not get a clean build

**Review by bartgol — APPROVED** [review]

**Pull request merged**
